modules/codegen-plugin/src/smithy4s/codegen/Smithy4sCodegenPlugin.scala

```diff
@@ -255,6 +263,24 @@ object Smithy4sCodegenPlugin extends AutoPlugin {
       val generateOptics = (config / smithy4sRenderOptics).value
       cached((wildcardArg, generateOptics))
     },
+    config / smithy4sGeneratedSmithyBuild := {
+      val baseDir = baseDirectory.value
+      val sourceManagedDir = (config / sourceManaged).value
+
+      val scalaBin = (config / scalaBinaryVersion).?.value
+      val deps = (config / libraryDependencies).value
+        .filter(_.configurations.exists(_.contains(Smithy4s.name)))
+        .flatMap(moduleIdEncode(_, scalaBin))
+      val imports = (config / smithy4sInputDirs).value
+        .collect(prepareInputDirs(baseDir, sourceManagedDir))
+      val repositories = externalResolvers.value.collect(prepareResolvers)
+      SmithyBuildJson.toJson(imports, deps, repositories)
+    },
+    config / smithy4sGeneratedSmithyBuildFile := {
+      val out = baseDirectory.value / "smithy-build.json"
+      IO.write(out, (config / smithy4sGeneratedSmithyBuild).value)
+      out
```

Comment:
not sure this cuts it : ideally we want to aggregate all dependencies from all modules and create the file from this aggregation.

Reply:
wish the LSP supported multiple config files in a workspace :/

contributions welcome I guess